std/assembly/array.ts

```diff
@@ -47,11 +47,13 @@ export class Array<T> {
     var buffer = allocateUnsafe(byteLength);
     this.buffer_ = buffer;
     this.length_ = length;
-    memory.fill(
-      changetype<usize>(buffer) + HEADER_SIZE,
-      0,
-      <usize>byteLength
-    );
+    if (byteLength) {
```

Comment:
Is this check necessary? Memory routines should handle zero length, which is uncommon anyway, well.

Reply:
Hmm, in my view case `new Array()` is pretty common

Reply:
Hmm, yeah, just noticed that this is the array constructor. Yet, this calls through to here, which already does the check: https://github.com/AssemblyScript/assemblyscript/blob/master/std/assembly/internal/memory.ts#L191

Reply:
ok, I'll remove this

Reply:
We could add this later, when mem bulk operations landed

Reply:
Yeah, if the intrinsic turns out to be slow for some reason. Otherwise, removing it for now at least prevents doing the same check twice in the `length > 0` case. Thanks :)